src/csharp/Extensions/Microsoft.Spark.Extensions.DotNet.Interactive/AssemblyKernelExtension.cs

```
@@ -0,0 +1,155 @@
+                        TryEmitAssembly(kernel, tempDir.FullName, out string assemblyPath))
+                    {
+                        sparkSession.SparkContext.AddFile(assemblyPath);
+
+                        foreach (string filePath in GetPackageFiles(tempDir.FullName))
+                        {
+                            sparkSession.SparkContext.AddFile(filePath);
+                        }
+                    }
+
+                    await next(command, context);
+                });
+            }
+
+            return Task.CompletedTask;
+        }
+
+        private DirectoryInfo CreateTempDirectory()
+        {
+            string envTempDir = Environment.GetEnvironmentVariable(TempDirEnvVar);
+            string tempDirBasePath = string.IsNullOrEmpty(envTempDir) ?
+                Directory.GetCurrentDirectory() :
+                envTempDir;
+
+            if (!IsPathValid(tempDirBasePath))
+            {
+                throw new Exception($"[{GetType().Name}] Spaces in " +
+                    $"'{tempDirBasePath}' is unsupported. Set the {TempDirEnvVar} " +
+                    "environment variable to control the base path. Please see " +
+                    "https://issues.apache.org/jira/browse/SPARK-30126 and " +
+                    "https://github.com/apache/spark/pull/26773 for more details");
+            }
+
+            return Directory.CreateDirectory(
+                Path.Combine(tempDirBasePath, Path.GetRandomFileName()));
+        }
+
+        private bool TryEmitAssembly(CSharpKernel kernel, string dstPath, out string assemblyPath)
+        {
+            Compilation compilation = kernel.ScriptState.Script.GetCompilation();
+            string assemblyName =
+                AssemblyLoader.NormalizeAssemblyName(compilation.AssemblyName);
+            assemblyPath = Path.Combine(dstPath, $"{assemblyName}.dll");
+            if (!File.Exists(assemblyPath))
```

I thought this was a critical error scenario no?

---

I don't think it's expected.   Should we log an error or throw an exception ?

---

let's throw an exception to fail fast